pkg/prometheus/promcfg.go

```diff
@@ -1346,14 +1346,14 @@ func generateRunningFilter() yaml.MapSlice {
 	}
 }
 
-func getLimit(user uint64, enforced *uint64) uint64 {
+func getLimit(user *uint64, enforced *uint64) uint64 {
 	if enforced != nil {
-		if user < *enforced && user != 0 || *enforced == 0 {
-			return user
+		if user != nil && *user < *enforced && *user != 0 || *enforced == 0 {
+			return *user
 		}
 		return *enforced
 	}
-	return user
+	return 0
```

Comment:
wouldn't it be better to return nil and the caller would set the field only if it's not nil?

Reply:
@ArthurSens Good point, if we now return a *uint64, we can do that.

Reply:
Updated in 17c9d21d9218a6f48a3c5918f735f5ea916a4adc